DependencyInjection/Configuration.php

```diff
@@ -25,6 +25,7 @@ public function getConfigTreeBuilder()
             ->booleanNode('throw_exception')->defaultFalse()->end()
             ->scalarNode('fallback_image')->defaultNull()->end()
             ->scalarNode('web_dir')->defaultValue('%kernel.root_dir%/../web')->end()
+            ->scalarNode('adapter')->defaultValue('gd')->end()
```

Comment:
This is not a class name and will not work.

I think it would be better to create services of adapters and configure the service name instead.

BTW, maybe can you write some tests using this library? https://packagist.org/packages/matthiasnoback/symfony-dependency-injection-test